packages/databricks-vscode/src/workspace-fs/WorkspaceFsAccessVerifier.ts

```diff
@@ -65,30 +85,38 @@ export class WorkspaceFsAccessVerifier implements Disposable {
     }
 
     async verifyCluster(cluster?: Cluster) {
-        if (
-            workspaceConfigs.syncDestinationType === "repo" ||
-            cluster === undefined
-        ) {
+        if (cluster === undefined) {
             return;
         }
-        const dbrVersionParts = cluster.dbrVersion;
-        if (
-            dbrVersionParts[0] < 11 ||
-            (dbrVersionParts[0] === 11 &&
-                dbrVersionParts[1] !== "x" &&
-                dbrVersionParts[1] < 2)
-        ) {
+        if (await dbrBelowThreshold(cluster)) {
+            if (workspaceConfigs.syncDestinationType === "repo [deprecated]") {
+                return;
+            }
             const message =
                 "Files in workspace is not supported on clusters with DBR < 11.2.";
             const selection = await window.showErrorMessage(
                 message,
                 "Switch to Repos",
                 "Ignore"
             );
-
             if (selection === "Switch to Repos") {
                 switchToRepos();
             }
+        } else {
+            if (workspaceConfigs.syncDestinationType === "workspace") {
+                return;
+            }
+            const message =
+                "Please switch to workspace for better experience, repos has been deprecated";
+            const selection = await window.showErrorMessage(
+                message,
+                "Switch to Workspace",
+                "Ignore"
+            );
+
+            if (selection === "Switch to Workspace") {
+                switchToWorkspace();
+            }
```

Comment:
For files in workspace to be enabled:
1. The cluster must be > 11.2
2. The files in workspace option must be enabled on workspace level. 

The 2nd case is not handled here. The way to check for 2nd case is to try uploading a file to the workspace and checking if it is able to upload correctly. This function does that https://github.com/databricks/databricks-vscode/blob/main/packages/databricks-vscode/src/workspace-fs/WorkspaceFsAccessVerifier.ts#L95. 

Can you please add handling for this case as well.